jsonpb: test failed with appengine build tags

test failed with appengine build tags.

go version:
```
$ go version
go version go1.10.3 linux/amd64
```

without build tags:
```
$ go test ./jsonpb/
ok      github.com/golang/protobuf/jsonpb       0.007s
```

with appengine build tags:
```
$ go test -tags=appengine ./jsonpb/
--- FAIL: TestMarshalAnyJSONPBMarshaler (0.00s)
panic: reflect.Value.Interface: cannot return value obtained from unexported field or method [recovered]
        panic: reflect.Value.Interface: cannot return value obtained from unexported field or method

goroutine 22 [running]:
testing.tRunner.func1(0xc4200ba4b0)
        /usr/local/go/src/testing/testing.go:742 +0x29d
panic(0x5da9a0, 0x663b20)
        /usr/local/go/src/runtime/panic.go:502 +0x229
reflect.valueInterface(0x5d1ce0, 0xc42007f800, 0x36, 0x1, 0xc420044c80, 0x4a128a)
        /usr/local/go/src/reflect/value.go:959 +0x1c1
reflect.Value.Interface(0x5d1ce0, 0xc42007f800, 0x36, 0xc42007f800, 0x1b8)
        /usr/local/go/src/reflect/value.go:948 +0x44
github.com/golang/protobuf/proto.pointer.toString(0x5d1ce0, 0xc42007f800, 0x36, 0x1b8)
        /root/gopath/src/github.com/golang/protobuf/proto/pointer_reflect.go:244 +0x3f
github.com/golang/protobuf/proto.sizeStringValueNoZero(0x5d1ce0, 0xc42007f800, 0x36, 0x1, 0x1)
        /root/gopath/src/github.com/golang/protobuf/proto/table_marshal.go:1229 +0x43
github.com/golang/protobuf/proto.(*marshalInfo).size(0xc420123340, 0x604a80, 0xc42007f800, 0x16, 0x40e18d)
        /root/gopath/src/github.com/golang/protobuf/proto/table_marshal.go:182 +0x12e
github.com/golang/protobuf/proto.(*InternalMessageInfo).Size(0xc42002ae40, 0x665140, 0xc42007f800, 0x0)
        /root/gopath/src/github.com/golang/protobuf/proto/table_marshal.go:124 +0xdd
github.com/golang/protobuf/proto.Marshal(0x665140, 0xc42007f800, 0x0, 0xf, 0x769d2b, 0x24, 0x2d2)
        /root/gopath/src/github.com/golang/protobuf/proto/table_marshal.go:2716 +0xf2
github.com/golang/protobuf/ptypes.MarshalAny(0x665140, 0xc42007f800, 0x2d2, 0x4d0d70, 0xc42007f701)
        /root/gopath/src/github.com/golang/protobuf/ptypes/any.go:66 +0x3c
github.com/golang/protobuf/jsonpb.TestMarshalAnyJSONPBMarshaler(0xc4200ba4b0)
        /root/gopath/src/github.com/golang/protobuf/jsonpb/jsonpb_test.go:587 +0x85
testing.tRunner(0xc4200ba4b0, 0x64a1d8)
        /usr/local/go/src/testing/testing.go:777 +0xd0
created by testing.(*T).Run
        /usr/local/go/src/testing/testing.go:824 +0x2e0
FAIL    github.com/golang/protobuf/jsonpb       0.007s
```

--- a/Makefile
+++ b/Makefile
@@ -36,6 +36,7 @@ install:
 
 test:
 	go test ./... ./protoc-gen-go/testdata
+	go test -tags purego ./... ./protoc-gen-go/testdata
 	go build ./protoc-gen-go/testdata/grpc/grpc.pb.go
 	make -C conformance test
 

--- a/jsonpb/jsonpb_test.go
+++ b/jsonpb/jsonpb_test.go
@@ -572,7 +572,7 @@ func TestMarshalIllegalTime(t *testing.T) {
 
 func TestMarshalJSONPBMarshaler(t *testing.T) {
 	rawJson := `{ "foo": "bar", "baz": [0, 1, 2, 3] }`
-	msg := dynamicMessage{rawJson: rawJson}
+	msg := dynamicMessage{RawJson: rawJson}
 	str, err := new(Marshaler).MarshalToString(&msg)
 	if err != nil {
 		t.Errorf("an unexpected error occurred when marshalling JSONPBMarshaler: %v", err)
@@ -583,7 +583,7 @@ func TestMarshalJSONPBMarshaler(t *testing.T) {
 }
 
 func TestMarshalAnyJSONPBMarshaler(t *testing.T) {
-	msg := dynamicMessage{rawJson: `{ "foo": "bar", "baz": [0, 1, 2, 3] }`}
+	msg := dynamicMessage{RawJson: `{ "foo": "bar", "baz": [0, 1, 2, 3] }`}
 	a, err := ptypes.MarshalAny(&msg)
 	if err != nil {
 		t.Errorf("an unexpected error occurred when marshalling to Any: %v", err)
@@ -601,7 +601,7 @@ func TestMarshalAnyJSONPBMarshaler(t *testing.T) {
 }
 
 func TestMarshalWithCustomValidation(t *testing.T) {
-	msg := dynamicMessage{rawJson: `{ "foo": "bar", "baz": [0, 1, 2, 3] }`, dummy: &dynamicMessage{}}
+	msg := dynamicMessage{RawJson: `{ "foo": "bar", "baz": [0, 1, 2, 3] }`, Dummy: &dynamicMessage{}}
 
 	js, err := new(Marshaler).MarshalToString(&msg)
 	if err != nil {
@@ -1011,8 +1011,8 @@ func TestUnmarshalJSONPBUnmarshaler(t *testing.T) {
 	if err := Unmarshal(strings.NewReader(rawJson), &msg); err != nil {
 		t.Errorf("an unexpected error occurred when parsing into JSONPBUnmarshaler: %v", err)
 	}
-	if msg.rawJson != rawJson {
-		t.Errorf("message contents not set correctly after unmarshalling JSON: got %s, wanted %s", msg.rawJson, rawJson)
+	if msg.RawJson != rawJson {
+		t.Errorf("message contents not set correctly after unmarshalling JSON: got %s, wanted %s", msg.RawJson, rawJson)
 	}
 }
 
@@ -1036,7 +1036,7 @@ func TestUnmarshalAnyJSONPBUnmarshaler(t *testing.T) {
 		t.Errorf("an unexpected error occurred when parsing into JSONPBUnmarshaler: %v", err)
 	}
 
-	dm := &dynamicMessage{rawJson: `{"baz":[0,1,2,3],"foo":"bar"}`}
+	dm := &dynamicMessage{RawJson: `{"baz":[0,1,2,3],"foo":"bar"}`}
 	var want anypb.Any
 	if b, err := proto.Marshal(dm); err != nil {
 		t.Errorf("an unexpected error occurred when marshaling message: %v", err)
@@ -1097,30 +1097,30 @@ func (s *stringField) UnmarshalJSONPB(jum *Unmarshaler, js []byte) error {
 // dynamicMessage implements protobuf.Message but is not a normal generated message type.
 // It provides implementations of JSONPBMarshaler and JSONPBUnmarshaler for JSON support.
 type dynamicMessage struct {
-	rawJson string `protobuf:"bytes,1,opt,name=rawJson"`
+	RawJson string `protobuf:"bytes,1,opt,name=rawJson"`
 
 	// an unexported nested message is present just to ensure that it
 	// won't result in a panic (see issue #509)
-	dummy *dynamicMessage `protobuf:"bytes,2,opt,name=dummy"`
+	Dummy *dynamicMessage `protobuf:"bytes,2,opt,name=dummy"`
 }
 
 func (m *dynamicMessage) Reset() {
-	m.rawJson = "{}"
+	m.RawJson = "{}"
 }
 
 func (m *dynamicMessage) String() string {
-	return m.rawJson
+	return m.RawJson
 }
 
 func (m *dynamicMessage) ProtoMessage() {
 }
 
 func (m *dynamicMessage) MarshalJSONPB(jm *Marshaler) ([]byte, error) {
-	return []byte(m.rawJson), nil
+	return []byte(m.RawJson), nil
 }
 
 func (m *dynamicMessage) UnmarshalJSONPB(jum *Unmarshaler, js []byte) error {
-	m.rawJson = string(js)
+	m.RawJson = string(js)
 	return nil
 }
 

